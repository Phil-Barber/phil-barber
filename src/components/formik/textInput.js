import React from 'react';
import { useField } from 'formik';
import styled from 'styled-components';

export const InputContainer = styled.div`
  min-height: 84px;
`;

export const Label = styled.label``;
export const Input = styled.input`
  width: 100%;
`;
export const Error = styled.div`
  color: red;
`;

export const TextInput = ({ label, className, textarea, ...props }) => {
  const [field, meta] = useField(props);
  const inputId = props.id || props.name;
  return (
    <InputContainer className={className}>
      <Label htmlFor={inputId}>{label}</Label>
      <Input
        id={inputId}
        as={textarea ? 'textarea' : 'input'}
        {...field}
        {...props}
      />
      {meta.touched && meta.error && (
        <Error className="error" data-testid="error">
          {meta.error}
        </Error>
      )}
    </InputContainer>
  );
};
