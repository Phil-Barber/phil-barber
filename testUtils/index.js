import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../src/style';
import '@testing-library/jest-dom/extend-expect';

const customRender = (ui, options, store) => {
  const AllTheProviders = ({ children }) => (
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
