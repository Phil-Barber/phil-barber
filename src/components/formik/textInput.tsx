import * as React from 'react';
import { useField, FieldConfig } from 'formik';
import styled from 'styled-components';

export const InputContainer = styled.div`
  min-height: 84px;
`;

export const Label = styled.label``;
export const Input = styled.input`
  width: 100%;
`;
export const Error = styled.div`
  color: red;
`;

interface Props extends FieldConfig {
  label: string;
  className?: string;
  textarea?: boolean;
  id?: string;
  placeholder?: string;
}

export const TextInput: React.FC<Props> = ({
  label,
  className,
  textarea,
  id,
  ...props
}: Props) => {
  const [field, meta] = useField(props);
  const inputId = id || props.name;
  return (
    <InputContainer className={className}>
      <Label htmlFor={inputId}>{label}</Label>
      <Input
        id={inputId}
        {...field}
        {...props}
        as={textarea ? 'textarea' : 'input'}
      />
      {meta.touched && meta.error && (
        <Error className="error" data-testid="error">
          {meta.error}
        </Error>
      )}
    </InputContainer>
  );
};
