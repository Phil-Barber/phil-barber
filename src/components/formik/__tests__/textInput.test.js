import React from 'react';
import { render, fireEvent } from '../../../../testUtils';
import { TextInput } from '../textInput';
import { Formik, Form } from 'formik';
import * as Yup from 'yup';

describe('TextInput', () => {
  const inputName = 'test-input';
  const label = 'test label';
  const defaultProps = {
    label,
    name: inputName,
  };

  let component;

  beforeEach(() => {
    component = render(
      <Formik
        initialValues={{
          [inputName]: '',
        }}
        validationSchema={Yup.object({
          [inputName]: Yup.string().required('Required'),
        })}
      >
        {() => (
          <Form>
            <TextInput {...defaultProps} />
            );
          </Form>
        )}
      </Formik>
    );
  });

  it('Matches snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });

  it('Shows error on error', async () => {
    fireEvent.blur(component.getByLabelText(label));
    expect(await component.findByTestId('error')).toBeInTheDocument();
  });
});
