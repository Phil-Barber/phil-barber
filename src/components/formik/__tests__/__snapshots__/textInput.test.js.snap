// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Matches snapshot 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="sc-AxjAm cKYGmA"
    >
      <label
        class="sc-AxirZ ktnnZK"
        for="test-input"
      >
        test label
      </label>
      <input
        class="sc-AxiKw hSLCUF"
        id="test-input"
        name="test-input"
        value=""
      />
    </div>
    );
  </form>
</div>
`;
